xarray/core/rolling.py

```diff
@@ -494,6 +527,14 @@ def _numpy_or_bottleneck_reduce(
                 bottleneck_move_func, keep_attrs=keep_attrs, **kwargs
             )
         else:
+            if fillna is not None:
+                if fillna is dtypes.INF:
+                    fillna = dtypes.get_pos_infinity(self.obj.dtype, max_for_int=True)
```

Comment:
useless since we always pad with NaN which ends up promoting to float. We should add `fill_value` support to `rolling`